src/transformers/models/bert/tokenization_bert.py

```diff
@@ -443,10 +448,12 @@ def _run_strip_accents(self, text):
             output.append(char)
         return "".join(output)
 
-    def _run_split_on_punc(self, text, never_split=None):
-        """Splits punctuation on a piece of text."""
+    def _split_on_punc_or_pattern(self, text, never_split=None):
+        """Splits a piece of text by self.pattern or punctuation."""
         if never_split is not None and text in never_split:
             return [text]
+        if self.pattern:
+            return re.findall(self.pattern, text, flags=re.UNICODE)
```

Comment:
We could add the following line in the class init:
`self.pattern = pattern if pattern else r"([\-\[\]\(\)\{\}\.,!?;:''^$`‘’“”\"/\\]+)"`

We then wouldn't need the `if self.pattern` conditional or anything below this line in this fn, but there could be something about splitting punctuation I'm missing here.

What do you think about this idea?